lib/go/test/ClientMiddlewareExceptionTest.thrift

```diff
@@ -25,11 +25,11 @@ exception Exception2 {
 
 // This is a special case, we want to make sure that the middleware don't
 // accidentally pull result as error.
-exception FooResponse {
+struct/*exception*/ FooResponse {      // returning an exception by any means other than "throws" is illegal
```

Comment:
the purpose of this test is to verify that the library code does not have a bug to accidentally pull the exception response out as the thrown exception, so this change will actually make the test always pass even if someone introduces such a bug in the future.

but if your plan is to forbid that on the compiler level so this scenario would never happen in the first place, then I think this is fine.

Reply:
I think I can construct the test differently (construct the go struct directly instead of via thrift IDL) to keep the test to do what it does, though.

Reply:
> I think I can construct the test differently [...] to keep the test to do what it does, though.

Great, thanks!

> if your plan is to forbid that on the compiler level 

I am actually thinking about this, indeed